File: no-issue-in-java/src/main/java/com/chrispie/MyLocalDateTimeAdapter.java
package com.chrispie;

import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE_TIME;

import java.time.LocalDateTime;

import jakarta.xml.bind.annotation.adapters.XmlAdapter;

// Not the best code but the question is not about using this class and not the implementation

public class MyLocalDateTimeAdapter extends XmlAdapter<String, LocalDateTime> {

    @Override
    public LocalDateTime unmarshal(String dateString) {
        return LocalDateTime.parse(dateString);
    }

    @Override
    public String marshal(LocalDateTime dateTime) {
        if (dateTime != null) {
            return dateTime.format(ISO_LOCAL_DATE_TIME);
        }
        return null;
    }

}

File: issue-in-kotlin/src/main/kotlin/com/chrispie/MyLocalDateTimeAdapter.kt
package com.chrispie

import jakarta.xml.bind.annotation.adapters.XmlAdapter
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

// Not the best code but the question is not about using this class and not the implementation

class MyLocalDateTimeAdapter : XmlAdapter<String?, LocalDateTime?>() {

    override fun unmarshal(dateString: String?): LocalDateTime? {
        return LocalDateTime.parse(dateString!!, DateTimeFormatter.ISO_LOCAL_DATE_TIME)
    }

    override fun marshal(dateTime: LocalDateTime?): String? {
        return dateTime?.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
    }
}

